ui/src/views/admin/settings/SettingsView.vue

```diff
@@ -1,82 +1,145 @@
+                      />
+                    </ValidationProvider>
+                  </v-col>
+                  <v-col
+                    cols="12"
+                    class="my-0 py-0"
+                  >
+                    <ValidationProvider
+                        v-slot="{ errors }"
+                        name="Password"
+                        rules="required|min:8"
+                    >
+                      <v-text-field
+                        v-model.trim="formData.password"
+                        outlined
+                        required
+                        hint="Minimum 8 characters"
+                        persistent-hint
+                        label="Password"
+                        class="my-0 py-0"
+                        :append-icon="isPasswordShown ? 'mdi-eye' : 'mdi-eye-off'"
+                        :type="isPasswordShown ? 'text' : 'password'"
+                        :error-messages="errors[0]"
+                        @click:append="isPasswordShown = !isPasswordShown"
+                      />
+                    </ValidationProvider>
+                  </v-col>
+                  <v-col
+                    cols="12"
+                  >
+                    <v-row justify="center">
+                      <v-btn
+                        color="primary"
+                        min-width="150"
+                        class="mt-2 mb-5"
+                        v-bind:disabled='!isDisabled'
+                        @click="handleSubmit(onFormSubmit)"
+                      >
+                        SAVE SETTINGS
+                      </v-btn>
+                    </v-row>
+                  </v-col>
+                </v-row>
+              </v-container>
+            </v-form>
+          </ValidationObserver>
+        </v-card>
+
+        <MaterialSnackbar
+          v-model="isSucessSnackbarShown"
+          type="success"
+        >
+          New settings have been saved!
+        </MaterialSnackbar>
       </v-col>
     </v-row>
   </v-container>
 </template>
 
 <script>
+import { mapGetters } from "vuex";
+
 export default { 
   name: "AdminSettingsView", 
   data() {
     return {
+      isFormValid: false,
+      isPasswordShown: false,
       formData: {
         email: "",
         password: ""
-      }
+      },
+      isSucessSnackbarShown: false,
     };
   },
   computed: {
-    profilePicture() {
-      return require("@assets/svg/profile_picture.svg");
-    }
-  } 
+    ...mapGetters("user", ["userData"]),
+    userEmail() {
+        return this.userData.email;
+    },
+    isDisabled() {
+        return this.formData.password.length > 7;
+    },
+  },
+  methods: {
+    onFormSubmit() {
+      // to do: post form data to back-end's update admin api
+      this.isSucessSnackbarShown = true;
+      setTimeout( () => this.$router.push({ name: "admin-surveys" }), 1000);
+    },
+  },
 };
 </script>
+
+<style lang="scss">
+:not(.v-select).v-text-field input[disabled="disabled"] {
```

Comment:
This selector is not good because it is too general; it may affect the other components rendered here  (try disabling the password field and I think with this selector, the pwd field will have the gray block as well). Additionally, I think you may have targeted the wrong selector because some parts of the text field are still not colored.

Reply:
I know that disabling the password field will also make it get grayed as well -- the idea of the styling is to make all disabled fields get grayed, isn't it? Nevertheless, I have removed this styling. I added a new color `disabled: "#B3B3B3"` to styles/variables.js and add `:background-color="style.color.disabled"` to the v-text-field. It looks better now because the entire element is now colored.

Reply:
This is also a good approach 👌